src/js/pm-image-editor.js

```diff
@@ -444,6 +448,11 @@
                     $scope.$broadcast('updateSelection', $scope.editorId, $scope.editor.selectionCss());
 
                     $scope.updateHistoryButtons();
+
+                    $scope.state = angular.copy($scope.editor.history.current());
+                    setTimeout(function() {
+                        $scope.$apply();
+                    }, 0);
```

Comment:
> I need this to avoid triggering issue that digest already in progress. Without this $apply state is not updated. I mean state is changing, but parent scope variable is not changing. You can doublecheck this behaviour if checkout current branch.
> 
> setTimeout 0 allows to move $apply out of execution chain

@antonvas it makes sense as you described, can you add some of your description as a comment? before that setTimeout so we remember next time why it was added there?